modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -336,6 +343,12 @@ template<typename Net> class Params {
         return *this;
     }
 
+    Params<Net>& cfgPreprocessingDeviceContext(void *device_ptr, void *context_ptr) {
+        desc.device_ptr = cv::util::make_optional(device_ptr);
```

Comment:
If we are leaving `device_selector` out of aboard - how we then specify device & context type?
In future on Win/Lin there will possible situation when we should support SYCL/ DX11/VA/OCL and others preprocessing together

I still recommend to reconsider  `IDeviceSelector` approach to do not brake GAPI API in future